lib/core/logger.cpp

```diff
@@ -0,0 +1,57 @@
+#include <spdlog/common.h>
+#include <spdlog/sinks/stdout_color_sinks.h>
+#include <spdlog/spdlog.h>
+#include <tl/expected.hpp>
+
+
+namespace {
+
+tl::expected<spdlog::level::level_enum, std::string> parseLoggerLevel(std::string levelStr) {
+    std::transform(levelStr.begin(), levelStr.end(), levelStr.begin(), [](unsigned char c) {
+        return std::tolower(c);
+    });
+
+    // TODO: magic_enum can hide these ifs
+    if (levelStr == "trace") {
```

Comment:
Please use `map` as we do not have `magic_enum`

Reply:
Indeed, Why didn't I think about map?